frontend/tests/e2e/tests/AddVisaResidentWithPHNTest.js

```diff
@@ -0,0 +1,156 @@
+import AddVisaResidentWithPHNPage from '../pages/enrollement/AddVisaResidentWithPHNPage';
+import AlertPage from '../pages/AlertPage';
+import PersonDetails from '../pages/enrollement/PersonDetailsPage';
+import { SITE_UNDER_TEST } from '../configuration';
+import { regularAccUser } from '../roles/roles';
+
+const immigrationCodeOption = AddVisaResidentWithPHNPage.immigrationCodeSelect.find('option');    
+const provinceOption = AddVisaResidentWithPHNPage.provinceSelect.find('option');
+const priorResidenceCodeOption = AddVisaResidentWithPHNPage.priorResidenceCodeInput.find('option');
+
+const ERROR_MESSAGE = 'Please correct errors before submitting';
+const SUCCESS_MESSAGE = 'HRPB242EINVALID SUBSCRIBER DATA FOUND. PLS CONTACT MSP.';
+const GROUPNUMBER_REQUIRED_MESSAGE = 'Group Number is required';
+const IMMIGRATION_CODE_REQUIRED_MESSAGE = 'Immigration Code is required';
+const PERMIT_ISSUE_DATE_REQUIRED_MESSAGE = 'Permit Issue Date is required';
+const PERMIT_EXPIRY_DATE_REQUIRED_MESSAGE = 'Permit Expiry Date is required';
+const RESIDENCE_DATE_REQUIRED_MESSAGE = 'Residence Date is required';
+const COVERAGE_EFFECTIVE_DATE_REQUIRED_MESSAGE = 'Coverage Effective Date is required';
+const COVERAGE_CANCELLATION_DATE_REQUIRED_MESSAGE = 'Coverage Cancellation Date is required';
+const HOME_ADDRESS_REQUIRED_MESSAGE = 'Home Address Line 1 is required';
+const CITY_REQUIRED_MESSAGE = 'City is required';
+const PROVINCE_REQUIRED_MESSAGE = 'Province is required';
+const POSTAL_CODE_REQUIRED_MESSAGE = 'Postal Code is required';
+const INVALID_GROUP_NUMBER_ERROR_MESSAGE = 'Group Number is invalid';
+const INVALID_GROUP_MEMBER_NUMBER_ERROR_MESSAGE = 'Group Member Number is invalid';
+const INVALID_DEPARTMENT_NUMBER_VALIDATION_MESSAGE = 'Department Number is invalid';
+const PHONE_NUMBER_VALIDATION_MESSAGE = 'Only numbers 0 to 9 are valid. Phone Number must be entered as ten (10) numbers in length with no space or hyphen.';
+
+const PAGE_TO_TEST = SITE_UNDER_TEST + '/coverage/enrollment/addVisaResidentWithPHN'
+
+fixture(`AddVisaResidentWithPHN Page`)
+.disablePageCaching `Test AddVisaResidentWithPHN`
+    .beforeEach(async t => {
+        await t
+            .useRole(regularAccUser)
+    })
+    .page(PAGE_TO_TEST);
+
```

Comment:
I believe there should be a test for required fields on the initial ResidentPHN page as well.